lib/MusicBrainz/Server/Data/Utils.pm

```diff
@@ -367,10 +367,16 @@ sub trim_multiline_text {
 
     $t = NFC($t);
     $t = remove_invalid_characters($t);
-    # Not trimming starting spaces to avoid breaking
-    # either list formatting in Wikitext
-    # or block in Markdown.
-    $t =~ s/\s+$//gm;
+
+    # Trimming each line to remove trailing spaces (or similar)
+    # - Not trimming starting spaces to avoid breaking
+    #   either list formatting in Wikitext
+    #   or block in Markdown.
+    # - Splitting on \n so that \s doesn’t match any \n
+    $t = join ("\n", map { $_ =~ s/\s+$//r } (split "\n", $t));
```

Comment:
In the initial commit you had `$t =~ s/[\t\r ]+$//gm;`, but all of the current tests pass with either piece of code. I guess that's to catch other characters matched by `\s` that aren't covered by `\t\r `?

Another option, if you prefer, is regex_sets, i.e. subtracting `\n` from `\s` like `/(?[ \s - \n ])/`. Though since it's experimental, you'll need to disable a warning: https://perldoc.perl.org/perlrecharclass#Extended-Bracketed-Character-Classes

Reply:
I tried many variations but I failed to make it work:
```perl
    $t =~ s/(?[ [\s] - [\n] ])$//gm;
```

Let’s not use experimental error-prone syntax when we have a single line code that works.

I added NBSP (U+00A0) to the tests though.

Reply:
The working variation is `$t =~ s/(?[ \s - \n ])+$//gm;`.

> I added NBSP (U+00A0) to the tests though.

Thanks!

Reply:
Plus I prefer to avoid using `m` for multi-line as it does not treat `\n` as I would expect.